### ISSUE TITLE
Issues when changing into the root directory

### ISSUE THREAD

**Issue by debo** [direnv/direnv#1134]
**Describe the bug**
I have two Apple Silicone MacBook Pro and on one I have no issues but on the second one I am experiencing a similar issue to [this other reported bug](https://github.com/direnv/direnv/issues/513).

It's worth noticing that the message is slightly different and the problem occurs only when I change into the root directory.

The message is
```
direnv: error can't find bash: exec: "bash": cannot run executable found relative to current directory
```

**To Reproduce**
Just run the following command
```shell
~ cd /
```

**Expected behavior**
I expect the shell to just change directory to the specified folder without throwing the aforementioned error message.

**Environment**
 - OS: MacOS Ventura 13.4.1 (22F82) on both systems
 - Shell: zsh on both systems but with bash available
 - Direnv version v2.32.3 on both systems

**Comment by zimbatm** [comment]
Can you run `type -p bash` on both computers and report back?

I would recommend `brew install bash` assuming you have Homebrew installed, as the Bash shipping with macOS is more than 12 years old!

**Comment by debo** [comment]
@zimbatm ok, surprisingly enough, the one that works without issue is the computer that uses the Bash shipped with MacOS, the one that uses the Homebrew one has the reported problem.

**Comment by zimbatm** [comment]
Thanks, can you give me the `echo $PATH` output for both?

One thing you can also try is to `export DIRENV_BASH=$(type -p bash)` and then see if things work better

**Comment by debo** [comment]
It would appear so, I'll post the two path out for completeness as soon as my internet comes back but I couldn't see anything obvious. Maybe you can spot something I didn't.

**Comment by zimbatm** [comment]
Look if there is some relative path like `.` in there

**Comment by debo** [comment]
Ok, so, [this](https://gist.github.com/debo/df85b2e482aca37e22bf50afa3ee6402) is the redacted (no username) PATH list of the system where it didn't work, and [this](https://gist.github.com/debo/adc0411e6b5afa80bc7f8d46a90f1f4d) s the path list. Both are in the right order.

As you can see the system where it didn't work has the following additional paths:
```
~/.rd/bin
~/.goenv/bin
~/.pub-cache/bin
/usr/local/share/dotnet
~/.dotnet/tools
/Library/Apple/usr/bin
~/.goenv/shim
```
Also, conversely, the one where it worked has this path that the other doesn't
```
/opt/homebrew/Cellar/libtool/2.4.7/libexec/gnubin
```

**Comment by zimbatm** [comment]
The usage of ~ in the PATH might be the cause of the issue. Try replacing `~/.rd/bin` with `$HOME/.rd/bin` for example, whenever it's set.

**Comment by zimbatm** [comment]
Ok I don't know then :)

**Comment by debo** [comment]
It's obviously down to the fact on one machine system bash has precedence or so it seems.

Interestingly enough running `bash --version` returns the same on both.

So something weird not just with paths.

**Comment by debo** [comment]
@zimbatm I just wanted to give you an update because I finally found out why it was working on one and not the other.

The one where it did work, it actually wasn't in the meaning that I didn't initialise `direnv` so it wasn't hooked to the shell. Once I did it both system had the same erroneous behaviour. The solution was to add the `export` you recommended as part of my environment initialisation.

Last but not least, I don't know why yet (and I probably won't investigate), but for some reason even though bash was is installed via `brew` on my systems, when I'm in the root directory `type -p bash` detects system `bash` rather than the `brew` one which is what led to the problem in the first place.

I hope this will help other people in future.
